### ISSUE TITLE
Trying to deploy on MKE (Docker Enterprise)

### ISSUE THREAD

**Issue by dajester2013** [arangodb/kube-arangodb#685]
I was successfully able to deploy the operator, but anytime I create an ArangoDeployment, I get this error trying to schedule the `-id` pod.

0/3 nodes are available: 1 node(s) had taint {com.docker.ucp.manager: }, that the pod didn't tolerate, 2 node(s) had taint {com.docker.ucp.orchestrator.kubernetes: }, that the pod didn't tolerate.

This is the manifest I'm trying to deploy:
```yaml
apiVersion: database.arangodb.com/v1alpha
kind: ArangoDeployment
metadata:
  name: dsmdb
spec:
  mode: Cluster
  image: arangodb/arangodb:3.6.1
  environment: Production
  tls:
    caSecretName: dsmdb-ca
  bootstrap:
    passwordSecretNames:
      root: dsmdb-credentials
```

I don't really want to go removing taints from nodes, as I'm leery of it messing with the MKE admin pods.

**Comment by ajanikow** [comment]
Hello!

You can set tolerations for your groups:

For ID: https://github.com/arangodb/kube-arangodb/blob/master/pkg/apis/deployment/v1/server_id_group_spec.go#L32
For other groups: https://github.com/arangodb/kube-arangodb/blob/master/pkg/apis/deployment/v1/server_group_spec.go#L63

Best Regards,
Adam JAnikowski.

**Comment by dajester2013** [comment]
Thanks, the first problem I found was that I needed to set the nodes in Docker EE to run either the kubernetes orchestrator, or in "mixed" mode.  They were previously set to "swarm".

However, I made the suggested changes:
```
apiVersion: database.arangodb.com/v1alpha
kind: ArangoDeployment
metadata:
  name: dsmdb
spec:
  mode: Cluster
  image: arangodb/arangodb:3.6.1
  environment: Production
  tls:
    caSecretName: dsmdb-ca
  bootstrap:
    passwordSecretNames:
      root: dsmdb-credentials
  agents:
    tolerations: 
    - key: com.docker.ucp.manager
      operator: Exists
      effect: "NoSchedule"
  coordinators:
    tolerations: 
    - key: com.docker.ucp.manager
      operator: Exists
      effect: "NoSchedule"
  dbservers:
    tolerations: 
    - key: com.docker.ucp.manager
      operator: Exists
      effect: "NoSchedule"
  syncmasters:
    tolerations: 
    - key: com.docker.ucp.manager
      operator: Exists
      effect: "NoSchedule"
  syncworkers:
    tolerations: 
    - key: com.docker.ucp.manager
      operator: Exists
      effect: "NoSchedule"
```

But it still results in an agent, a coordinator, and a prmr(?) to be unschedulable on the manager node:

```
0/3 nodes are available: 1 node(s) had taint {com.docker.ucp.manager: }, that the pod didn't tolerate, 2 node(s) didn't match pod affinity/anti-affinity, 2 node(s) didn't satisfy existing pods anti-affinity rules.
```

**Comment by ajanikow** [comment]
Hello!

You are missing tolerations for ID container (first link).

PRMR - it is DBServer (Primary) group.

Best Regards,
Adam Janikowski.

**Comment by ajanikow** [comment]
Hello!

As I can see you have 2 nodes, you cant run in this case cluster in Production environment, please switch to Development environment: https://www.arangodb.com/docs/stable/deployment-kubernetes-deployment-resource.html#specenvironment-string

Best Regards,
Adam Janikowski.

**Comment by dajester2013** [comment]
Thanks for the input - I'm working remotely this week, and perhaps into
next week, but I'll try once I get back.

It is actuallu a 3 node system - one of the nodes is the docker ee manager
nodes, which has a taint on it that isn't being tolerated, so 2 instances
of each service are coming online, but none are being scheduled on the
third.

**Comment by ajanikow** [comment]
Hello!

Can you share taints which are present there? Also, tolerations in rendered pods.

Best,
Adam.

**Comment by dajester2013** [comment]
I did try to patch the node and remove the com.docker.ucp.manager:NoSchedule taint, but UCP added it back immediately.

**Comment by ajanikow** [comment]
Hello!

I do not see toleration in tolerations list.

I tried something like:
```
apiVersion: "database.arangodb.com/v1"
kind: "ArangoDeployment"
metadata:
  name: "deployment"
spec:
  mode: Cluster
  environment: Production
  image: arangodb/enterprise:3.6.6
  imagePullPolicy: Always
  dbservers:
    tolerations:
      - key: "com.docker.ucp.manager"
        operator: "Exists"
```

And if I describe pod:
```
Tolerations:     com.docker.ucp.manager op=Exists
                 node.alpha.kubernetes.io/unreachable:NoExecute op=Exists for 300s
                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason                  Age   From                     Message
  ----    ------                  ----  ----                     -------
```

Can you check if you are using latest Operator?

Best Regards,
Adam.

**Comment by dajester2013** [comment]
> Can you check if you are using latest Operator?

I'm using arangodb/kube-arangodb:1.1.3

I copied your manifest, but it references the enterprise edition.  Are tolerations not supported for the community edition?

**Comment by ajanikow** [comment]
Hello!

Operator does not have limitations for Pod definitions based on edition. Same thing will work on Community (I just copied yaml from QA tests which test tolerations).

Did you test yaml which was provided by me?

Also, can you test Operator 1.1.5?

Best,
Adam.

**Comment by dajester2013** [comment]
Adam,

I had to stop work on this project for a while, but am now coming back around to it.

On further research, it seems MKE (UCP) is preventing the operator from adding that toleration to the pods.  I verified that adding any other toleration to the ArangoDeployment spec is coming through correctly, it's just that one particular toleration is not.

https://capstonec.com/2019/04/15/kubernetes-tolerations-working-together-with-docker-ucp-scheduler-restrictions/

What I need to find out, and have not been able to find so far, is how to configure the arango-deployment-operator service account so that it can schedule on the manager / dtr nodes.
